tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_reducers.ts

```diff
@@ -674,6 +675,102 @@ const reducer = createReducer(
       };
     }
   ),
+  ////////////////////////////////////////////////
+  // Reducers related to graph structures.      //
+  ////////////////////////////////////////////////
+  on(
+    actions.graphOpFocused,
+    (
+      state: DebuggerState,
+      data: {graph_id: string; op_name: string}
+    ): DebuggerState => {
+      return {
+        ...state,
+        graphs: {
+          ...state.graphs,
+          focusedOp: {
+            graphId: data.graph_id,
+            opName: data.op_name,
+          },
+        },
+      };
+    }
+  ),
+  on(
+    actions.graphOpInfoRequested,
+    (
+      state: DebuggerState,
+      data: {graph_id: string; op_name: string}
+    ): DebuggerState => {
+      const {graph_id, op_name} = data;
+      const newState: DebuggerState = {
+        ...state,
+        graphs: {
+          ...state.graphs,
+          loadingOps: {
+            ...state.graphs.loadingOps,
+          },
+        },
+      };
+      if (newState.graphs.loadingOps[graph_id] === undefined) {
+        newState.graphs.loadingOps[graph_id] = {};
+      }
+      if (newState.graphs.loadingOps[graph_id][op_name] === undefined) {
+        newState.graphs.loadingOps[graph_id][op_name] = DataLoadState.LOADING;
+      }
+      return newState;
+    }
+  ),
+  on(
+    actions.graphOpInfoLoaded,
+    (state: DebuggerState, data): DebuggerState => {
+      const opInfo: GraphOpInfo = data.graphOpInfoResponse;
+      const {inputs, consumers, graph_ids} = opInfo;
+      const graphId = graph_ids[graph_ids.length - 1];
+      const newState: DebuggerState = {
+        ...state,
+        graphs: {
+          ...state.graphs,
+          ops: {
+            ...state.graphs.ops,
+          },
+          loadingOps: {
+            ...state.graphs.loadingOps,
+          },
+        },
+      };
+      for (const input of inputs) {
+        if (!input.data) {
+          continue;
+        }
```

Comment:
Help lazy person like me by documenting how this can happen.

Reply:
Added a comment here. Basically, `input.data` can be undefined if the backend fails to lookup the data regarding the input op, e.g., due to certain special TF-internally generated ops such as `StatefulPartitionedCall`. This is explained in the documentation of the backend code: https://github.com/tensorflow/tensorboard/blob/master/tensorboard/plugins/debugger_v2/debugger_v2_plugin.py#L317